src/preferences.js

```diff
@@ -228,14 +243,40 @@ var CPUPowerPreferences = class CPUPowerPreferences {
             this._syncOrdering();
         }
 
+        // update profile context with given profile
+        profileContext.Profile.Name = profile.Name;
+        profileContext.Profile.MinimumFrequency = profile.MinimumFrequency;
+        profileContext.Profile.MaximumFrequency = profile.MaximumFrequency;
+        profileContext.Profile.TurboBoost = profile.TurboBoost;
+        profileContext.Profile.DefaultAC = profile.DefaultAC;
+        profileContext.Profile.DefaultBat = profile.DefaultBat;
+
         profileContext.Settings.NameEntry.set_text(profileContext.Profile.Name);
         profileContext.Settings.MinimumFrequencyScale.set_value(profileContext.Profile.MinimumFrequency);
         profileContext.Settings.MaximumFrequencyScale.set_value(profileContext.Profile.MaximumFrequency);
         profileContext.Settings.TurboBoostSwitch.set_active(profileContext.Profile.TurboBoost);
+        profileContext.Settings.AutoSwitchACCheck.set_active(profileContext.Profile.DefaultAC);
+        profileContext.Settings.AutoSwitchBatCheck.set_active(profileContext.Profile.DefaultBat);
         profileContext.ListItem.NameLabel.set_text(profileContext.Profile.Name);
         profileContext.ListItem.MinimumFrequencyLabel.set_text(profileContext.Profile.MinimumFrequency.toString());
         profileContext.ListItem.MaximumFrequencyLabel.set_text(profileContext.Profile.MaximumFrequency.toString());
         profileContext.ListItem.TurboBoostStatusLabel.set_text(profileContext.Profile.TurboBoost ? _("Yes") : _("No"));
+        if (profileContext.Profile.DefaultAC && profileContext.Profile.DefaultBat)
+        {
+            profileContext.ListItem.AutoSwitchConfigLabel.set_text("[" + _("AC") + "/" + _("Battery") + "]");
```

Comment:
We could use symbolic icons here for displaying which profile is currently the default battery and AC profile:

`battery-symbolic` (available via gtk icon-names)
![battery-symbolic](https://user-images.githubusercontent.com/5300274/74646954-15fdc000-517b-11ea-8397-7683a84f6284.png)

`ac-adapter-symbolic` (available via gtk icon-names)
![ac-adapter-symbolic](https://user-images.githubusercontent.com/5300274/74647100-58270180-517b-11ea-9d21-3aa6fdbb54e7.png)

Reply:
That was also my first intention, but I was not sure if these icons are available in every gnome installation. Glade listed them under non-standard icons.

Reply:
I think glade is wrong here. The documentation states that standard icons have a symbolic version of them with the `symbolic` suffix https://developer.gnome.org/hig/stable/icons-and-artwork.html.en

Also, a gnome developer on reddit answered this question: https://www.reddit.com/r/gnome/comments/978l7y/what_happened_to_symbolic_icons_in_gladegtk_why/

I think we can use the symbolic icons as they are part of the Adwaita icon theme which is installed with every GTK installation.

Reply:
Oh, and if the icons will be problem, we can just include the two icons as png's in our code :grin:

Reply:
Ok nice